examples/TailorMyResume.ipynb

```diff
@@ -0,0 +1,402 @@
+      "Requirement already satisfied: six>=1.5 in c:\\users\\manul\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from python-dateutil>=2.7->matplotlib) (1.16.0)\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip install requests\n",
+    "!pip install python-dotenv\n",
+    "!pip install matplotlib"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**02. Setting Up Environment Variables** \n",
+    "\n",
+    "Since you need to set the INSTILL_API_TOKEN, you can create a .env file to store the API key securely, and then load it in the notebook.\n",
+    "\n",
+    "Example .env file:\n",
+    "```plaintext\n",
+    "INSTILL_API_TOKEN=********\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import base64\n",
+    "import json\n",
+    "import requests\n",
+    "# Load environment variables from .env file\n",
+    "from dotenv import load_dotenv\n",
+    "import os\n",
+    "\n",
+    "# Ensure the .env file is in the same directory as this notebook\n",
+    "load_dotenv()\n",
+    "\n",
+    "INSTILL_API_TOKEN = os.getenv(\"INSTILL_API_TOKEN\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**03. Define Helper Functions to Interact with the API**\n",
+    "\n",
+    "You can write helper functions that will handle the file upload and API interaction. Here's an example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def submit_resume_to_api(resume_file_path, job_description):\n",
+    "    # API endpoint\n",
+    "    url = \"https://api.instill.tech/v1beta/users/manulthanura/pipelines/tailormyresume/trigger\"\n",
+    "\n",
+    "    # Headers for authorization\n",
+    "    headers = {\n",
+    "        \"Content-Type\": \"application/json\",\n",
+    "        \"Authorization\": f\"Bearer {INSTILL_API_TOKEN}\"\n",
+    "    }\n",
+    "\n",
+    "    # Load the resume file content and encode it as base64\n",
+    "    with open(resume_file_path, 'rb') as file:\n",
+    "        resume_content = base64.b64encode(file.read()).decode('utf-8')\n",
+    "\n",
+    "    # Payload to send to the API\n",
+    "    data = {\n",
+    "        \"inputs\": [\n",
+    "            {\n",
+    "                \"job_description\": job_description,\n",
+    "                \"resume_pdf_file\": resume_content\n",
+    "            }\n",
+    "        ]\n",
+    "    }\n",
+    "\n",
+    "    # Sending the POST request to the API\n",
+    "    response = requests.post(url, headers=headers, json=data)\n",
+    "\n",
+    "    if response.status_code == 200:\n",
+    "        return response.json()\n",
+    "    else:\n",
+    "        print(f\"Error: {response.status_code}, {response.text}\")\n",
+    "        return None\n"
```

Comment:
Please use our Python SDK to call pipelines. See other notebooks and our documentation for examples

Reply:
Yeah sure I will

Reply:
Yes, currently we only support Python 3.11. It will be upgraded to Python 3.12 soon, but to use our SDK you have to downgrade to 3.11